### ISSUE TITLE
Extra life/s • 1up ?

### ISSUE THREAD

**Issue by blauefuesse** [reloZid/android-anuto#66]
Wouldn't it be a nice feature to recieve an extra life every so often? I was thinking i.e. after reaching a certain level or even using a great sum of credit to obtain an extra life.
Thanx for this very coool game!!

**Comment by ralienpp** [comment]
Indeed, I think it would be fair, because it makes the game a bit more forgiving - a player can compensate for the mistakes they made when they were "young and stupid". The current approach is quite punishing - the moment you realize you screwed up, there's no point in continuing.

I never interrupted a game because of this, but when you're eventually defeated, there are regrets, of course.


So the question to ask is - what are the circumstances in which one earns a life?

**Comment by betsythefc** [comment]
I would like a new life every 20 or 25 levels. I think that would help getting to level 50.

**Comment by ralienpp** [comment]
I think that it is worth discussing the possibility of adding a difficulty level. With `easy` you get a life every 5 waves, whereas with `difficult` it is every 25 waves.

For me, the progress with the game had a very dramatic beginning, I couldn't get past level 10 no matter what I tried. If you give a life every 20 levels, it will be absolutely useless to the newbies.

I think the "1up" philosophy should be such that it gives a slight boost, but doesn't guarantee you'll get beyond your current bottleneck unless there is a paradigm shift in the strategy.



Background: after reading the issues here I saw people discussing "level 56" and I figured that level 10 is actually "passable", so I began trying unusual things that I never attempted in the past - eventually figuring out what a better approach is.

**Comment by reloZid** [comment]
Personally I hit the next wave button very early to maximize the bonus. This means that there are always a lot of enemies in game. If my defense breaks I almost always loose not because of one enemy, but because of like 20. So for me even an extra life every 5 levels would not help that much. You don't have that problem?

I like the idea of making the amount of lives more dynamically, but I would couple it with a decision the player has to make. The easiest one is to spend money for lives. What about a special tower? Like a graveyard which gives you a bonus life for every X enemies who died within its range?

**Comment by jkufner** [comment]
> If my defense breaks I almost always loose not because of one enemy, but because of like 20.

Exactly. I think it is not a big deal to reset lives every wave back to 5.

It would be much better to start again from the previously completed level. It would leave more space for experiments.

**Comment by ralienpp** [comment]
> because of one enemy, but because of like 20

I can relate to that. But I must also point out that a typical pattern is that because of some foolish mistake in the beginning I lose N-1 lives, but decide to keep playing. Often it happens that this game turns out to be quite successful and I am able to get very far, hanging on to that single life. Some time later I might lose because of a single enemy (not 20), thus I blame myself for not quitting earlier - all the time after that was wasted.


A game should be somewhat forgiving, pushing the player to the limit of their skills and a bit over, but not by too much. Right now the punishment is being forced to start from scratch. I don't know what the best approach is though, but we could summarize what was suggested so far and think of which idea is the fairest.

1. set lives to 5 after each wave
2. get an extra life every N waves
3. buy a life
4. `graveyard tower` 


I think items 3 and 4 have better prospects, because they allow players with better skills to get better results; whereas the first two items will treat all players equally, regardless of their performance.


A graveyard tower could consider the speed at which enemies are destroyed (kill fewer swiftly and get the same result as killing many over a longer period of time). I cannot predict the impact on the game balance overall - it will certainly spawn a new breed of strategies.


An alternative to the `graveyard` concept would be a `life sucker` - in some quest games "dark force wizards" can use a spell that takes the life out of their opponents to recharge manna - maybe that would be appropriate.

**Comment by valborg** [comment]
Is something happening with this?

**Comment by reloZid** [comment]
Currently the following has first priority for me:
* Fix bugs from v0.2
* Add in-game help
* Save game state if application is closed

So currently I'm not working on this. I'll focus on such things if I'm done with the basic issues. Of course everyone who likes is invited to contribute. But please tell me first so that we can decide how exactly this new tower works.

**Comment by gloryroad1** [comment]
I agree that bonus lives should be awarded and at a rate that makes them adequate compensation for one's acheivements and meaningful to one's game play.

I also agree that when I lose, that level often is an abrupt massacre where having gained less that 30 or 40 bonus lives wouldn't mean much.

Perhaps on the topic of compensation for acheivement, bonus credits that are adequate and meaningful would be more helpful.  Also, the idea of a tower that compensates one for the volume of slaughter in it's range sounds fun...I've created some impressive killing zones.

**Comment by zz-ha-dum** [comment]
awarded for surviving isn't enough.  There ought be an excellence metric.

**Comment by a-raccoon** [comment]
I would suggest 1 extra life for each level cleared. The basis for this being that death is usually a ridiculous devastation anyway where you go from having all 20 lives to -40 in just one wave.  It's always a steep cliff, would like to die at the skin of my teeth, after losing 100 lives in the last 2 waves.
